### PR TITLE
feat: fetch ERC20 tokens

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [x] I have followed the contribution guidelines for this project
-   [x] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/features/developer-tools.features.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ const developerToolsFeatures: IDeveloperFeatures = {
     deeplink: {
         enabled: true,
     },
+    importErc20Token: {
+        enabled: true,
```
Comment (MarkNerdi) [inline packages/desktop/features/developer-tools.features.ts:24]: is this already ready to be enabled?

File: packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { getSelectedAccountWrappedNativeTokens } from '.'
+import { getSelectedAccountWrappedNativeTokens } from './getSelectedAccountWrappedNativeTokens'
 
-export async function getSelectedAccountLayer2Tokens(): Promise<{ amount: bigint; id: string }[]> {
+export async function getSelectedAccountLayer2Tokens(): Promise<unknown[]> {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts:3]: why? we know the return type  because of line 51-54 in `getSelectedAccountWrappedNativeTokens`

File: packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
     const wrappedNativeTokens = await getSelectedAccountWrappedNativeTokens()
-    return wrappedNativeTokens
+    return [...wrappedNativeTokens]
```
Comment (MarkNerdi) [inline packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts:5]: why is this necessary?

Comment (maxwellmattryan) [reply packages/shared/lib/core/layer-2/actions/getSelectedAccountLayer2Tokens.ts:5]: Was remnant of a change for adding ERC20 stuff, will remove

File: packages/shared/lib/core/layer-2/actions/importErc20Token.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+export async function importErc20Token(tokenAddress: string, chainId: number): Promise<unknown> {
+    const chain = get(network)?.getChain(chainId)
+    const contract = chain?.getContract(ContractType.Erc20, tokenAddress)
+    if (contract) {
+        // TODO: Extract into separate function later
+        const name = await contract.methods.name().call()
+        const symbol = await contract.methods.symbol().call()
+        const decimals = await contract.methods.decimals().call()
+        console.log('METADATA: ', name, symbol, decimals)
+
```
[the remark this answers is not in the document]
Comment (maxwellmattryan) [reply packages/shared/lib/core/layer-2/actions/importErc20Token.ts:19]: Yes, that would be good :+1:

File: packages/desktop/features/developer-tools.features.ts (hunk 1 above)
Comment (Tuditi) [reply packages/desktop/features/developer-tools.features.ts:24]: For now, yes. It is in Developer Tools anyways and we need it to work for a release.

**Review by MarkNerdi — APPROVED** [review]
LGTM
